services/apis/indexd/indexdTasks.js

```diff
@@ -82,31 +86,77 @@ module.exports = {
    * @param {Object} fileNode - Assumed to have a did property
    * @returns {Promise<Gen3Response>}
    */
-  async getFile(file) {
+  async getFile(file, headers = user.mainAcct.accessTokenHeader) {
     // get data from indexd
     return I.sendGetRequest(
       `${indexdProps.endpoints.get}/${file.did}`,
-      user.mainAcct.accessTokenHeader,
+      headers,
     ).then((res) => {
       file.rev = getRevFromResponse(res);
       return res.body;
     });
   },
 
+  /**
+   * Updates indexd data for file
+   * @param {Object} fileNode - Assumed to have a did property
+   * @returns {Promise<Gen3Response>}
+   */
+  async updateFile(file, data, headers = user.mainAcct.accessTokenHeader) {
+    return I.sendGetRequest(
+      `${indexdProps.endpoints.get}/${file.did}`,
+      headers,
+    ).then((res) => {
+      // get last revision
+      file.rev = getRevFromResponse(res);
+
+      const headers = user.mainAcct.indexdAuthHeader;
```

Comment:
why do you use `mainAcct`? does `indexdAuthHeader` require the user to be an admin? if so, could you add a comment to explain this (either at `indexdAuthHeader` or here or both)

Reply:
ah, this is a mistake, should use indexdAuthHeader by default and allow overriding to user token. will fix. auth header is just basic auth with username/password and doesn't actually rely on a specific user (it just pulls from env vars).